PhysicsTools/PatAlgos/plugins/CaloJetSlimmer.cc

```diff
@@ -0,0 +1,63 @@
+
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "DataFormats/Common/interface/RefVector.h"
+
+#include "CommonTools/UtilAlgos/interface/StringCutObjectSelector.h"
+#include "CommonTools/UtilAlgos/interface/SingleObjectSelector.h"
+#include "CommonTools/UtilAlgos/interface/ObjectSelector.h"
+#include "CommonTools/UtilAlgos/interface/SingleElementCollectionSelector.h"
+
+#include "DataFormats/JetReco/interface/CaloJet.h"
+#include "DataFormats/PatCandidates/interface/Jet.h"
+#include "DataFormats/Candidate/interface/Candidate.h"
+
+#include <vector>
+
+class CaloJetSlimmer : public edm::stream::EDProducer<> {
+
+public:
+
+CaloJetSlimmer( edm::ParameterSet const & params ):
+    srcToken_(consumes<edm::View<reco::CaloJet> >( params.getParameter<edm::InputTag>("src") )),
+    cut_( params.getParameter<std::string>("cut") ),
+    selector_( cut_ )
+    {
+        produces< reco::CaloJetCollection> ();
+    }
+   
+    virtual ~CaloJetSlimmer() {}
+   
+    virtual void produce(edm::Event& iEvent, const edm::EventSetup& iSetup) override {
+        
+        auto caloJets = std::make_unique<reco::CaloJetCollection>();
+        
+        edm::Handle< edm::View<reco::CaloJet> > h_jets;
+        iEvent.getByToken( srcToken_, h_jets);
+
+        for (auto const& ijet : *h_jets){
+           if( selector_(ijet) ){
+              reco::CaloJet::Specific tmp_specific;
+              const reco::Candidate::Point orivtx(0,0,0);
```

Comment:
Hi @slava77, yes, the vertex coordinates are set to 0 in order to save space. We can set the the vertices later in the analysis. (Actually, in the displaced jet analysis, the calo jets should be assigned to secondary vertices instead of the PV).